patroni/dcs/consul.py

```diff
@@ -82,6 +82,9 @@ def wrapper(callback, path, params=None, data=''):
                 kwargs['timeout'] = (float(params['wait'][:-1]) if 'wait' in params else 300) + 1
             else:
                 kwargs['timeout'] = self._read_timeout
+            if isinstance(params, dict) and 'token' in params and params['token']:
+                kwargs['headers'] = {'X-Consul-Token': params['token']}
```

Comment:
lines 85-86 could be rewritten as:
```python
kwargs['headers'] = {'X-Consul-Token': params.pop('token')}
```

My main concern was about breaking compatibility with the old Consul versions, but it seems that 'X-Consul-Token' support was added in version 0.6.0, released on December 3, 2015, so probably it should be fine.

Reply:
Using `.pop` could generate an attribute error, as it is possible for `params` to be `None`

Reply:
Previous line guaranties that 'token' is defined and even non empty

Reply:
Got you, so the change should be on line 86 then. Will update